core/decoders/tight.js

```diff
@@ -398,12 +411,31 @@ export default class TightDecoder {
         return this._scratchBuffer;
     }
 
+    async _disableQOIWorkers() {
+        if (this._workers) {
+            this._enableQOI = false;
+            this._availableWorkers = null;
+            this._sabs = null;
+            this._sabsR = null;
+            this._arrs = null;
+            this._arrsR = null;
+            this._qoiRects = null;
+            this._rectQlooping = null;
+            for await (let i of Array.from(Array(this._threads).keys())) {
+                this._workers[i].terminate();
+                delete this._workers[i];
+            }
+            this._workers = null;
+        }
+    }
+
     _enableQOIWorkers() {
         let fullPath = window.location.pathname;
         let path = fullPath.substring(0, fullPath.lastIndexOf('/')+1);
         let sabTest = typeof SharedArrayBuffer;
+        let enablePassed = false;
         if (sabTest !== 'undefined') {
-            this._enableQOI = true;
+            enablePassed = true;
```

Comment:
It's a matter of style but personally I think less indentation makes the code easier to read so I would return as soon as possible:

```
const supportsSharedArrayBuffers = typeof SharedArrayBuffer !== "undefined";
If (!supportsSharedArrayBuffers) {
    return false;
}

...

return true;
```

Reply:
Agreed and fixed.